### ISSUE TITLE
Postgres url does not load

### ISSUE THREAD

**Issue by nlhkh** [playframework/play-slick#324]
In my play app I have this config

```hocon
slick.dbs.default = {
  driver = "slick.driver.PostgresDriver$"
  db = {
    dataSourceClass = slick.jdbc.DatabaseUrlDataSource
    properties = {
      driver = org.postgresql.Driver
      url = "postgres://username:password@host:port/dbname"
    }
  }
}
```

However, I am getting an error where it wants `slick.dbs.default.db.url`:
`com.typesafe.config.ConfigException$Missing: No configuration setting found for key 'db.url'`

I was told that with Play-Slick 1.1.0, I can use postgres url instead of jdbc url, and settung `dataSourceClass` will skip `slick.dbs.default.db.url`, but obviously it is not the case.

**Comment by dotta** [comment]
@nlhkh It's a bug, and here is the root's cause https://github.com/playframework/play-slick/blob/master/src/evolutions/src/main/scala/play/api/db/slick/evolutions/internal/DBApiAdapter.scala#L54

**Comment by nlhkh** [comment]
@dotta thanks for pointing out it's a bug :). I was thinking that I missed the configuration somewhere.

**Comment by dotta** [comment]
Nope, problem is on our side, and should be really easy to fix. Out of curiosity, would you have the full exception stacktrace?

**Comment by nlhkh** [comment]
Here it is
```play.api.UnexpectedException: Unexpected exception[PSQLException: ERROR: relation "play_evolutions" does not exist
  Position: 64]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:165) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:121) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at scala.Option.map(Option.scala:146) ~[scala-library-2.11.7.jar:na]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1.apply(DevServerStart.scala:121) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1.apply(DevServerStart.scala:119) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at scala.util.Success.flatMap(Try.scala:231) ~[scala-library-2.11.7.jar:na]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1.apply(DevServerStart.scala:119) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1.apply(DevServerStart.scala:111) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[scala-library-2.11.7.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[scala-library-2.11.7.jar:na]
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402) ~[na:1.8.0_60]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) ~[na:1.8.0_60]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) ~[na:1.8.0_60]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) ~[na:1.8.0_60]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) ~[na:1.8.0_60]
Caused by: org.postgresql.util.PSQLException: ERROR: relation "play_evolutions" does not exist
  Position: 64
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2182) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1911) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:173) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Statement.java:616) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeWithFlags(AbstractJdbc2Statement.java:452) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeQuery(AbstractJdbc2Statement.java:332) ~[postgresql-9.4-1204-jdbc42.jar:9.4]
	at com.zaxxer.hikari.proxy.StatementProxy.executeQuery(StatementProxy.java:99) ~[HikariCP-java6-2.3.7.jar:na]
	at com.zaxxer.hikari.proxy.StatementJavassistProxy.executeQuery(StatementJavassistProxy.java) ~[HikariCP-java6-2.3.7.jar:na]
	at play.api.db.evolutions.DatabaseEvolutions.executeQuery(EvolutionsApi.scala:298) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.DatabaseEvolutions.databaseEvolutions(EvolutionsApi.scala:125) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.DatabaseEvolutions.scripts(EvolutionsApi.scala:97) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.DatabaseEvolutions.scripts(EvolutionsApi.scala:112) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.DefaultEvolutionsApi.scripts(EvolutionsApi.scala:77) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions$$anonfun$play$api$db$evolutions$ApplicationEvolutions$$runEvolutions$1.apply$mcV$sp(ApplicationEvolutions.scala:50) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions.withLock(ApplicationEvolutions.scala:98) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions.play$api$db$evolutions$ApplicationEvolutions$$runEvolutions(ApplicationEvolutions.scala:49) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions$$anonfun$start$1.apply(ApplicationEvolutions.scala:42) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions$$anonfun$start$1.apply(ApplicationEvolutions.scala:42) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at scala.collection.immutable.Stream.foreach(Stream.scala:594) ~[scala-library-2.11.7.jar:na]
	at play.api.db.evolutions.ApplicationEvolutions.start(ApplicationEvolutions.scala:42) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutions.<init>(ApplicationEvolutions.scala:149) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get$lzycompute(EvolutionsModule.scala:53) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get(EvolutionsModule.scala:53) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get(EvolutionsModule.scala:44) ~[play-jdbc-evolutions_2.11-2.4.0.jar:2.4.0]
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81) ~[guice-4.0.jar:na]
	at com.google.inject.internal.BoundProviderFactory.provision(BoundProviderFactory.java:72) ~[guice-4.0.jar:na]
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61) ~[guice-4.0.jar:na]
	at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:62) ~[guice-4.0.jar:na]
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103) ~[guice-4.0.jar:na]
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40) ~[guice-4.0.jar:na]
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:145) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:205) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:199) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:199) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:180) ~[guice-4.0.jar:na]
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:110) ~[guice-4.0.jar:na]
	at com.google.inject.Guice.createInjector(Guice.java:96) ~[guice-4.0.jar:na]
	at com.google.inject.Guice.createInjector(Guice.java:73) ~[guice-4.0.jar:na]
	at com.google.inject.Guice.createInjector(Guice.java:62) ~[guice-4.0.jar:na]
	at play.api.inject.guice.GuiceBuilder.injector(GuiceInjectorBuilder.scala:126) ~[play_2.11-2.4.3.jar:2.4.3]
	at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:93) ~[play_2.11-2.4.3.jar:2.4.3]
	at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:21) ~[play_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1$$anonfun$2.apply(DevServerStart.scala:153) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1$$anonfun$2.apply(DevServerStart.scala:150) ~[play-server_2.11-2.4.3.jar:2.4.3]
	at play.utils.Threads$.withContextClassLoader(Threads.scala:21) ~[play_2.11-2.4.3.jar:2.4.3]
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:150) ~[play-server_2.11-2.4.3.jar:2.4.3]
	... 14 common frames omitted
```

The bug seems to be related to the creation of `play_evolutions` table. Perhaps when it try to create that table, it has the wrong config. Once I have my `play_evolutions` table created, any subsequent update to the schema is picked up without problem with my config as above.

**Comment by dotta** [comment]
Thanks a lot! Yes, it appears the `url` is needed by the evolutions module to create the `play_evolutions` table. I'll have a closer look at it. In the meanwhile, to workaround the issue you could create the `play_evolutions` table yourself.

**Comment by nlhkh** [comment]
Thanks!

**Comment by dotta** [comment]
@nlhkh I've pushed a PR with a fix for this, it's in #327. Would you mind trying it out and confirm it addresses the problem. If that's the case, I'll happily release 1.1.1.

**Comment by nlhkh** [comment]
@dotta How can I try out the fix? I suppose I should git clone play-slick as a subproject of my Play project?

**Comment by nlhkh** [comment]
I tried various way to use this PR, including building the jars directly, publish to local ivy, and use sbt child project, but they all have problem integrating with my Play project, specifically there is no `JdbcProfile`. If you could publish a SNAPSHOT to maven, then maybe I can check it out.

**Comment by dotta** [comment]
Hey @nlhkh, here are the steps for building my branch and publishing the JARs locally:

```
git clone https://github.com/dotta/play-slick.git
cd play-slick
git checkout issue/324-url-exception
sbt +publishLocal
```

The version should be `1.2.0-SNAPSHOT`, and you will need to update your project's build to use that.

You should see the JARs in your local ivy folder, e.g., `~/.ivy2/local/com.typesafe.play/play-slick-evolutions_2.10/1.2.0-SNAPSHOT/`

**Comment by nlhkh** [comment]
I just tested it out, and I confirm that this fix works!

**Comment by dotta** [comment]
@nlhkh Awesome. Thanks for testing!

**Comment by dotta** [comment]
@nlhkh I've released 1.1.1 - see #329

**Comment by nlhkh** [comment]
Thanks!
